crates/core/src/network/grpc/connector.rs

```diff
@@ -20,22 +20,23 @@ use super::MAX_MESSAGE_SIZE;
 use crate::network::net_util::create_tonic_channel;
 use crate::network::protobuf::core_node_svc::core_node_svc_client::CoreNodeSvcClient;
 use crate::network::protobuf::network::Message;
-use crate::network::{NetworkError, ProtocolError, TransportConnect};
+use crate::network::transport_connector::find_node;
+use crate::network::{ConnectError, TransportConnect};
 
-#[derive(Clone)]
-pub struct GrpcConnector;
+#[derive(Clone, Default)]
+pub struct GrpcConnector {
+    _private: (),
+}
 
 impl TransportConnect for GrpcConnector {
     async fn connect(
         &self,
         node_id: GenerationalNodeId,
         nodes_config: &NodesConfiguration,
         output_stream: impl Stream<Item = Message> + Send + Unpin + 'static,
-    ) -> Result<
-        impl Stream<Item = Result<Message, ProtocolError>> + Send + Unpin + 'static,
-        NetworkError,
-    > {
-        let address = nodes_config.find_node_by_id(node_id)?.address.clone();
+    ) -> Result<impl Stream<Item = Message> + Send + Unpin + 'static, ConnectError> {
+        // todo: remove this unwrap when we start passing address as input
```

Comment:
Do you mean with `unwrap` the `?` operator?

Reply:
No, it's a leftover from a previous iteration.